### ISSUE TITLE
get_world_from_local_position uses incorrect quaternion/vector embedding and returns wrong world positions when rotated

### ISSUE THREAD

**Issue by zopenge** [isaac-sim/IsaacSim#414]
### Description

Description
The helper function get_world_from_local_position in isaacsim.core.utils.python.impl.torch.transformations appears to be mathematically incorrect and produces wrong world positions once the body is rotated.
File:
source/extensions/isaacsim.core.utils/python/impl/torch/transformations.py

@torch.jit.script
def get_world_from_local_position(pos_offset_local: torch.Tensor, pose_global: torch.Tensor):
    """Convert a point from the local frame to the global frame
    Args:
        pos_offset_local: Point in local frame. Shape: [N, 3]
        pose_global: The spatial pose of this point. Shape: [N, 7]
    Returns:
        Position in the global frame. Shape: [N, 3]
    """
    quat_pos_local = torch.cat(
        [
            pos_offset_local,
            torch.zeros(pos_offset_local.shape[0], 1, dtype=torch.float32, device=pos_offset_local.device),
        ],
        dim=-1,
    )
    quat_global = pose_global[:, 3:7]
    quat_global_conj = quat_conjugate(quat_global)
    pos_offset_global = quat_mul(quat_global, quat_mul(quat_pos_local, quat_global_conj))[:, 0:3]

    result_pos_gloal = pos_offset_global + pose_global[:, 0:3]

    return result_pos_gloal

What’s wrong
In rotations.py, quaternions are consistently treated as [w, x, y, z] (scalar first):

# rotations.py
w1, x1, y1, z1 = a[:, 0], a[:, 1], a[:, 2], a[:, 3]
# ...
quat = torch.stack([w, x, y, z], dim=-1)

Expected behavior
get_world_from_local_position(pos_offset_local, pose_global) should be equivalent to:

pos = pose_global[:, 0:3]
quat = pose_global[:, 3:7]  # [w, x, y, z]
pos_offset_global = quat_apply(quat, pos_offset_local)
return pos + pos_offset_global

and must preserve the distance to the body origin and the correct local direction for any rotation.

### Isaac Sim version

2.1.0

### Operating System (OS)

win11

### GPU Name

2080Ti

### GPU Driver and CUDA versions

CUDA version: 12.8

**Comment by vick-yu** [comment]
An internal ticket has been created to address this issue and we’re tracking it internally. Thanks for the clear report and detailed analysis.
​

**Comment by vick-yu** [comment]
Hi @zopenge,                                                                                                                                                                                                                                       
                                                                                                                                                                                                                                                    
 This has been fixed in Isaac Sim 6.0.0. The get_world_from_local_position function now uses quat_apply(quat, pos_offset_local) instead of the manual quaternion sandwich multiplication that had the incorrect [x, y, z, 0] pure quaternion        
 embedding (should have been [0, x, y, z] for scalar-first convention).                                                                                                                                                                             
                                                                                                                                                                                                                                                    
 We verified the fix produces correct results for identity rotations, 90° axis rotations, and preserves distance under arbitrary rotations — exactly as your report described.                                                                      
                                                                                                                                                                                                                                                    
 Thanks for the clear analysis and suggested fix!

**Comment by vick-yu** [comment]
Closing.
